File: single-row-keyboard.js
/**
 * @param {string} keyboard
 * @param {string} word
 * @return {number}
 */
var calculateTime = function(keyboard, word) {
  let time = 0;
  previousIndex = 0
  let currentIndex;
  const arr = word.split('');
  arr.forEach((letter) => {
    // console.log(keyboard.indexOf(letter));
    currentIndex = keyboard.indexOf(letter);
    time += Math.abs(currentIndex - previousIndex);
    previousIndex = currentIndex;
  });
  return time;
};

const keyboard = 'pqrstuvwxyzabcdefghijklmno';
const word = 'leetcode';

calculateTime(keyboard, word);

File: single-row-keyboard.kt
import kotlin.math.abs

class Solution {
  fun calculateTime(keyboard: String, word: String): Int {
    var time = 0
    var previousIndex = 0
    for (i in 0..word.length - 1) {
      var index = keyboard.indexOf(word[i])
      time += Math.abs(index - previousIndex)
      previousIndex = index
    }
    return time
  }
}